chain/client/src/client.rs

```diff
@@ -2140,3 +2151,52 @@ impl Client {
         }
     }
 }
+
+impl Client {
+    fn get_tier1_accounts(&mut self, tip: &Tip) -> Result<Arc<AccountKeys>, Error> {
+        if let Some(it) = self.tier1_accounts.get(&tip.epoch_id) {
+            return Ok(it.clone());
+        }
+        let info = self
+            .runtime_adapter
+            .get_validator_info(ValidatorInfoIdentifier::EpochId(tip.epoch_id.clone()))
+            .map_err(|e| Error::ChainWithCtx("runtime.get_validator_info()", e))?;
+        let mut accounts = BTreeMap::new();
+        accounts.extend(
+            info.current_validators
+                .into_iter()
+                .map(|v| ((tip.epoch_id.clone(), v.account_id), v.public_key)),
+        );
+        accounts.extend(
+            info.next_validators
+                .into_iter()
+                .map(|v| ((tip.next_epoch_id.clone(), v.account_id), v.public_key)),
+        );
+        let accounts = Arc::new(accounts);
+        self.tier1_accounts.put(tip.epoch_id.clone(), accounts.clone());
+        Ok(accounts)
+    }
+
+    fn send_network_chain_info(&mut self) -> Result<(), Error> {
```

Comment:
I'd maybe passed the `tip` in. Right now, the signature looks like it can be called whenever. With the `tip` in, it's a hint that this should be called when we upgdade the tip. `new_tip: Tip` arg would I think improve understandability quite a bit here.

Reply:
This will require head() call error handling outside of the function. As you can see at the call site the error is just logged (and I don't really know why it is fallible in the first place, I'm just replicating the error-tolerant code already present in the near_client). This either means wrapping the call sited in yet-another-function (possibly a closure) so that it can early exit if head() call fails, or having an ugly nested "if" with 2 logging calls, one per each fallible call. Which one you have in mind?

Reply:
Yeah, good point! In theory, we should already have the `tip` at the call-site, eg from `check_and_update_doomslug_tip`, but that'd require a bigger refactor.